File: 백준/Silver/2606. 바이러스/바이러스.java
import java.util.*;
import java.io.*;

public class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(reader.readLine());
        int M = Integer.parseInt(reader.readLine());
        List<List<Integer>> graph = new ArrayList<>();
        for(int i = 0; i <= N;i++) {
            graph.add(new ArrayList<>());
        }    

        for(int i = 0; i < M; i++){
            String[] input = reader.readLine().split(" ");
            int x = Integer.parseInt(input[0]);
            int y = Integer.parseInt(input[1]);
            graph.get(x).add(y);
            graph.get(y).add(x);
        }
        boolean visited[] = new boolean[N + 1];
        Queue<Integer> q = new LinkedList<>();
        q.add(1);
        visited[1] = true;
        int answer = 0;
        while(!q.isEmpty()) {
            int cur = q.poll();
            for(int neighbor : graph.get(cur)){
                if(!visited[neighbor]){
                    visited[neighbor] = true;
                    q.add(neighbor);
                    answer++;
            }
        }
    }
        System.out.println(answer);
    }
}

File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <iostream>
#include <vector>
#include <queue>

using namespace std;

int main() {
    int N, M;
    cin >> N >> M;

    vector<vector<int>> graph(N + 1); 

    for (int i = 0; i < M; i++) {
        int x, y;
        cin >> x >> y; 
        graph[x].push_back(y);
        graph[y].push_back(x);
    }

    vector<bool> visited(N + 1, false);
    queue<int> q;
    q.push(1); 
    visited[1] = true;
    int answer = 0;

    while (!q.empty()) {
        int cur = q.front();
        q.pop();
        for (int neighbor : graph[cur]) {
            if (!visited[neighbor]) {
                visited[neighbor] = true;
                q.push(neighbor);
                answer++;
            }
        }
    }

    cout << answer << endl;
    return 0;
}
